### DISCUSSION TITLE
Raise the rarity of "complex" ICs

### DISCUSSION THREAD

**Discussion by k-7iro** [MrTJP/ProjectRed#1902] [Ideas]
In the real world, the more complex and powerful something is, the more expensive it tends to be, just like Core i9 is more expensive than Core i3. However, in Project Red, there is no reason to make chips such as RAM and adders smaller. In the real world, low-grade CPUs almost always use the same die as high-grade CPUs, and are made by disabling defective dies. So, in the same way, in Project Red, the more components are placed in a chip, the more likely it is to produce defective products. This makes it worthwhile to make RAM, adders, etc. smaller or to disable expansion functions when making them in Project Red. In survival servers, chips with higher performance will be more expensive and more realistic.
We propose `70-(Tile Count/5)%` as one formula for calculating the production probability. In the image below, the production probability is 28%, 41%, and 54%. If you feel it is too low, you can change 70 to 80 or 5 to a higher number.
![image](https://github.com/user-attachments/assets/4b9f00a4-a2d5-418b-8ad9-7b2afd368710)
![image](https://github.com/user-attachments/assets/66a41d36-8033-43c5-903d-d684a73a4d06)
![image](https://github.com/user-attachments/assets/d99cd833-cc01-4da3-a3a6-7a8473a5a536)
![image](https://github.com/user-attachments/assets/822c6f26-b36d-4019-916a-391eafecfe7a)
![image](https://github.com/user-attachments/assets/16e27496-1bec-4290-a663-e71a7b268534)
![image](https://github.com/user-attachments/assets/bf224e3c-099d-4325-b0f6-f3882bdf2eea)

**Comment by MrTJP** [comment]
Hello! Yea I hear you. We actually do have a scaling mechanism where a defect has a chance of occurring for every unit area. Therefore, you need to shrink the process node to have a better chance at yield. Some of this is explained in the top "info" tab in the IC Workbench.

This mechanism hasn't been balanced yet though, so it's totally possible that large IC's are not "rare" enough, as you said. However, note that the layering feature is not accessible yet in game, and the defect chance exponentially increases the more layers you add (i.e. there must be no defects calculated at any of the layers).

Our current defect formula has the following components:
1. The [wafer type](https://github.com/MrTJP/ProjectRed/blob/main/fabrication/src/main/java/mrtjp/projectred/fabrication/lithography/WaferType.java#L6) used, which determines the base "chance of defect per unit are"
2. The [process node](https://github.com/MrTJP/ProjectRed/blob/main/fabrication/src/main/java/mrtjp/projectred/fabrication/lithography/ProcessNode.java#L4), which determines the area of a single tile in your IC (The current IC size is hard-coded to 16x16 tiles)
3. The number of layers in your design. The defect calculation is run on every layer. For example, 2 layers will have half the chance of being defect-free.

The system currently does not take into account the number of tiles actually used though. That's an interesting idea.
